### DISCUSSION TITLE
DataTable: Cell navigation like Excel Spreadsheet

### DISCUSSION THREAD

**Discussion by 1Map** [primefaces/community#1360] [PrimeVue]
In the DataTable, how can I navigate through cells using keyboard (arrow keys), like in, for example: MS Excel

I want to highlight the focused row (color1) as well as the focused cell (color2).  And then, when I use my keyboard arrow keys, navigate from one cell to the next.

left / right = stay on focused row and move between columns
up /down = move focus between rows, but stay on focused column
home, end, page up, page down should also behave like Excel

Like in example: https://www.primefaces.org/showcase-ext/sections/sheet/basicUsage.jsf

**Comment by 1Map** [comment]
Any feedback on this?

**Comment by msladek** [comment]
This PR https://github.com/primefaces/primevue/pull/6349 was just rejected

**Comment by mergehez** [comment]
Since my previous [PR](https://github.com/primefaces/primevue/pull/6349) was rejected, I’d like to suggest a different approach: adding `keyboard-navigation` option. If set to true, it would enable the same behavior I had proposed in my PR. 

However, one issue with this is that keyboard-navigation would always be available, even if `edit-mode` isn’t set to "cell." That could lead to some confusion.

Another possible solution: adding a new edit mode like "navigated-cell" (or something similar). This way, the keyboard navigation would only kick in when this specific mode is enabled, keeping it separate from the existing modes.

I really want this to be implemented, so please let me know what you think or if there's a better way to handle this.
